File: Basic/JS/1013.js
/**
 * @author: kurumi
 * @date: 2025-11-08
 * @description: 1013 数素数
 */
const rl = require("readline").createInterface({input: process.stdin});
const iter = rl[Symbol.asyncIterator]();
const readline = async () => (await iter.next()).value;

const primeIndex = {
  start: 0,
  end: 0
};
const MAX = 110000;
const primes = new Array(MAX).fill(true);

function countPrimes(m, n) {
  let cnt = 0;
  primes[1] = false;
  primes[2] = true;
  for (let i = 2; i < primes.length; i++) {
    if (primes[i]) {
      ++cnt;
      if (cnt == m) primeIndex.start = i;
      if (cnt == n) {
        primeIndex.end = i;
        break;
      }
      for (let j = i; i * j <= MAX; j++) primes[i * j] = false;
    }
  }
}

void (async function () {
  const [m, n] = (await readline()).split(" ").map(Number);
  countPrimes(m, n);
  const rowMax = 10;
  let cnt = 0;
  for (let i = primeIndex.start; i <= primeIndex.end; i++) {
    if (primes[i]) {
      if (cnt % 10 == 0 && cnt != 0) {
        console.log();
      } else if (cnt % 10 != 0) {
        process.stdout.write(" ");
      }
      ++cnt;
      process.stdout.write(i.toString());
    }
  }
})();

File: Basic/Python/1013.py
'''
@Author: kurumi
@Date: 2025-11-08
@Description: 1013 数素数
'''
from typing import List

MAX: int = 110000

def countPrimes(primes: List[bool], m: int, n: int):
    cnt, start, end = 0, 0, 0
    primes[1], primes[2] = False, True
    for i in range(2, len(primes)):
        if primes[i] == True:
            cnt += 1
            if cnt == m:
                start = i
            if cnt == n:
                end = i
                break
            j: int = i
            while i * j < MAX:
                primes[i * j] = False
                j += 1
    return (start, end)

def main():
    [m, n] = map(int, input().split())
    rowMax, cnt = 10, 0
    primes: List[bool] = [True] * MAX
    (start, end) = countPrimes(primes, m, n)
    for i in range(start, end + 1):
        if primes[i]:
            if cnt % 10 == 0 and cnt != 0:
                print()
            elif cnt % 10 != 0:
                print(" ", end="")
            cnt += 1
            print(i, end="")

if __name__ == '__main__':
    main()